ios/Classes/PTFlutterDocumentController.m

```diff
@@ -0,0 +1,728 @@
+
+- (void)annotationViewControllerDidCancel:(PTAnnotationViewController *)annotationViewController
+{
+    [annotationViewController dismissViewControllerAnimated:YES completion:nil];
+}
+
+#pragma mark - Viewer Settings
+
+- (void)initViewerSettings
+{
+    _hideAnnotationToolbarSwitcher = NO;
+    _hideTopToolbars = NO;
+    _hideTopAppNavBar = NO;
+    _readOnly = NO;
+    
+    _showNavButton = YES;
+}
+
+- (void)applyViewerSettings
+{
+    // nav icon
+    [self applyNavIcon];
+    
+    const BOOL hideNav = (self.hideTopAppNavBar || self.hideTopToolbars);
+    self.controlsHidden = hideNav;
+    
+    const BOOL translucent = hideNav;
+    self.navigationController.navigationBar.translucent = translucent;
+    self.thumbnailSliderController.toolbar.translucent = translucent;
+    
+    // Always enable the bottom toolbar. This is required for the custom checks in -controlsHidden
+    // when the top toolbar(s) are disabled, to be able to still toggle the bottom toolbar.
+    self.bottomToolbarEnabled = YES;
+    
+    // Always allow toggling toolbars on tap.
+    BOOL hidesToolbarsOnTap = YES;
+    self.hidesControlsOnTap = hidesToolbarsOnTap;
+    self.pageFitsBetweenBars = !hidesToolbarsOnTap; // Tools default is enabled.
+    
+    [self applyDocumentControllerSettings];
+}
+
+- (void)applyNavIcon
+{
+    if (self.showNavButton) {
+        UIBarButtonItem* navButton = navButton = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone target:self action:@selector(topLeftButtonPressed:)];
+        
+        self.leadingNavButtonItem = navButton;
+        
+        NSArray<UIBarButtonItem *> *compactItems = [self.navigationItem leftBarButtonItemsForSizeClass:UIUserInterfaceSizeClassCompact];
+        if (compactItems) {
+            NSMutableArray<UIBarButtonItem *> *mutableItems = [compactItems mutableCopy];
+            [mutableItems insertObject:navButton atIndex:0];
+            compactItems = [mutableItems copy];
+        } else {
+            compactItems = @[navButton];
+        }
+        [self.navigationItem setLeftBarButtonItems:compactItems
+                                            forSizeClass:UIUserInterfaceSizeClassCompact
+                                                animated:NO];
+        
+        NSArray<UIBarButtonItem *> *regularItems = [self.navigationItem leftBarButtonItemsForSizeClass:UIUserInterfaceSizeClassRegular];
+        if (regularItems) {
+            NSMutableArray<UIBarButtonItem *> *mutableItems = [regularItems mutableCopy];
+            [mutableItems insertObject:navButton atIndex:0];
+            regularItems = [mutableItems copy];
+        } else {
+            regularItems = @[navButton];
+        }
+        [self.navigationItem setLeftBarButtonItems:regularItems
+                                            forSizeClass:UIUserInterfaceSizeClassRegular
+                                                animated:NO];
+    }
+}
+
+- (void)applyDocumentControllerSettings
```

Comment:
What is "DocumentControl"? This seems to be all related to toolbars/groups, should that name be used instead?